src/cmd/ksh93/tests/sh_match.sh

```diff
@@ -411,12 +411,131 @@ function test_nomatch
     [[ $got == "$expect" ]] || log_error 'unset submatches not handled correctly'
 }
 
-# run tests
+# ========
+# This is a regexp to extract pattern to pattern entries from related API unit test source. It emits
+# the patterns in a form this script can easily use; i.e., the two patterns separated by a tab.
+# See the next two test functions.
+p2p_sep=$'\t'
+p2p_re=$'/ tests\\[\\] =/,/NULL, NULL/s/^.*{"\\(.*\\)", "\\(.*\\)"},$/\\1\t\\2/p'
+# These patterns are known not to be correctly converted to a glob. We still test them but only
+# issue a warning rather than failing the test. See https://github.com/att/ast/issues/1367.
+typeset -A ere2glob_blacklist
+ere2glob_blacklist['^x\!y$']=1
+ere2glob_blacklist['^x|y$']=1
+ere2glob_blacklist['^x\a|b$']=1
+# These patterns are known not to be correctly converted to a ere. We still test them but only
+# issue a warning rather than failing the test. See https://github.com/att/ast/issues/1367.
+typeset -A glob2ere_blacklist
+glob2ere_blacklist['x!y']=1
+glob2ere_blacklist['x|y']=1
+glob2ere_blacklist['~(E)z.?a']=1
+glob2ere_blacklist['x\a|b']=1
+
+# ========
+# This is related to unit test API/string/fmtmatch. The difference is this uses ksh to convert the
+# pattern then uses ksh to verify both patterns match the same lines.
+function test_ere2glob {
+    # Generate a list of files from the source tree of the project.
+    # Then generate a file containing some text from those files.
+    find "$SRC_ROOT/src" -type f > ere2glob.files
+    : > ere2glob.txt
```

Comment:
Why do you need a separate step to create this file ?

Reply:
Because it's moderately expensive and both functions need some randomly generated text. I originally had this in each test function but when I noticed how expensive it is (especially on Cygwin) decided to hoist it to cut the cost in half.